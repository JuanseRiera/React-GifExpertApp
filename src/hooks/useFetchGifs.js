import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = (category) => {
  const [state, setstate] = useState({ data: [], loading: true });

  useEffect(() => {
    getGifs(category).then((respuesta) => {
      setstate({
        data: respuesta,
        loading: false,
      });
    });
  }, [category]);

  return state;
};
