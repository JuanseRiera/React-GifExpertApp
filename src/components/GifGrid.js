import React from "react";
import PropTypes from "prop-types";
import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {
  let { data, loading } = useFetchGifs(category);

  const htmlToRender = () => {
    if (!loading && data && data.length > 0) {
      return (
        <div className="cardContainer">
          {data?.map((img) => {
            return <GifGridItem key={img.id} {...img} />;
          })}
        </div>
      );
    }
    if (!loading && (!data || data.length === 0)) {
      return (
        <>
          <p className="nombreCategoria animate__animated animate__headShake">
            No se encontraron elementos para la busqueda realizada.
          </p>
        </>
      );
    }
  };
  return (
    <>
      <h3 className="nombreCategoria animate__animated animate__flipInY">
        {category}
      </h3>
      {loading && (
        <p className="nombreCategoria animate__animated animate__flash">
          Cargando...
        </p>
      )}
      {htmlToRender()}
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
