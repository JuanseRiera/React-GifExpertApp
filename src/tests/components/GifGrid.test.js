import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs");

describe("Pruabs del componente gifgrid", () => {
  test("Debería de cargar el componente", () => {
    let mockData = { data: [], loading: true };
    useFetchGifs.mockReturnValue(mockData);
    const wrapper = shallow(<GifGrid category={"Sherlock"} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debería de cargar el componente", () => {
    let mockData = {
      data: [
        {
          id: "1",
          url: "https://google.com",
          title: "Hola mundo",
        },
      ],
      loading: false,
    };
    useFetchGifs.mockReturnValue(mockData);
    const wrapper = shallow(<GifGrid category={"Sherlock"} />);
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(mockData.data.length);
  });
});
