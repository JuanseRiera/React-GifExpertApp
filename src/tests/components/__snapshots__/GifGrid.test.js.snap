// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruabs del componente gifgrid Deber√≠a de cargar el componente 1`] = `
<Fragment>
  <h3
    className="nombreCategoria animate__animated animate__flipInY"
  >
    Sherlock
  </h3>
  <p
    className="nombreCategoria animate__animated animate__flash"
  >
    Cargando...
  </p>
</Fragment>
`;
